pkg/cluster/mpi/statusupdate.go

```diff
@@ -130,11 +149,27 @@ func getPods(ctx *core.Context, cr *dcv1alpha1.MPICluster) ([]corev1.Pod, error)
 	return podList.Items, err // first item is an empty array when err == nil
 }
 
-func isPodReady(pod corev1.Pod) bool {
+// isPodReady determines is the given Pod is in a "ready" state with an optional
+// string describing why it's not ready.
+func isPodReady(pod corev1.Pod) (ready bool, reason string) {
 	for _, cond := range pod.Status.Conditions {
 		if cond.Type == corev1.PodReady {
-			return cond.Status == corev1.ConditionTrue
+			if cond.Status == corev1.ConditionTrue {
+				return true, ""
+			}
+			break
+		}
+	}
+	for _, containerStatus := range pod.Status.ContainerStatuses {
```

Comment:
Should we look at specific containers by name? 

There may be sidecars that we aren't thinking about like Istio proxy.

Reply:
Yes, but I intentionally handle a sidecar failure and a main container failure in the same way. I don't think we should distinguish them, as a non-functional rsync is equally bad.